transport/tchannel/header.go

```diff
@@ -36,14 +36,14 @@ import (
 
 // pullBaggage pulls the context headers from the given transport.Headers,
 // deleting them from the original headers map.
-func pullBaggage(headers transport.Headers) transport.Headers {
-	ctxHeaders := make(transport.Headers)
+func pullBaggage(headers transport.Headers) map[string]string {
```

Comment:
This returns a `map[string]string` because `baggage.NewContextWithHeaders` expects one. That expects a map because it's just an internal bulk version of `baggage.NewContext(ctx, string, string)`. And since it has to construct a new Headers object regardless, it will end up normalizing all the keys again anyway.